collect_app/src/test/java/org/odk/collect/android/utilities/TextUtilsTest.java

```diff
@@ -0,0 +1,33 @@
+package org.odk.collect.android.utilities;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+
+@RunWith(RobolectricTestRunner.class)
+public class TextUtilsTest {
+
+    /**
+     * Should return null if provided with null and not throw a NPE.
+     */
+    @Test
+    public void textToHtml_BounceNullInput() {
+        String input = null;
+        CharSequence observed = TextUtils.textToHtml(input);
+        assertNull(observed);
+    }
+
+    /**
+     * Should do something not-null-ish if provided with non-null string.
+     */
+    @Test
+    public void textToHtml_CallsHTMLOnRealInput() {
+        String input = "This *bold* markdown is \n- nice,\n- efficient.\n";
+        CharSequence result = TextUtils.textToHtml(input);
+        assertNotNull(result);
```

Comment:
I LOVE that we are adding tests, and a huge kudos to you for doing so. I feel like with the change from the original commit (which had the `captor` business), however, that this method name is no longer appropriate. Right now it's just asserting that the returned result is not null, not that it actually calls the value on the input. Someone could rewrite `textToHtml` to just return an empty string and this test would still pass.

I think that a better test for this would be something like:

```java
@Test
public void textToHtml_CallsHTMLOnRealInput() 
    String arg = "bar";
    String expected = "supposedly styled string";
    TextUtils tuMock = mock(TextUtils.class);
    when(tuMock.textToHtml(eq(arg))).thenReturn(expected);
    String actual = tuMock.textToHtml(arg);
    verify(tuMock, times(1)).returnFoo(eq("baz"));
    assertEquals(actual, expected);
}
```

There I'm trying to assert both that the method was called with the argument expected and that we are returning the result of the `textToHtml` method. The unfortunate thing is that all the methods are static as written. I agree with your original sentiment that this is ugly. I'd vote that these methods either change to non-static and we test like above, or we just change the name of the test to better describe what it's actually doing.

Adding a test framework is a big win, and I'd LGTM a PR like this that lays the groundwork for more tests even if it doesn't add a lot of tests itself.